### PR TITLE
backend/remote: do not unlock after a failed upload

### PR DESCRIPTION
**Pull request by svanharmelen**
When changes are made and we failed to upload the state, we should not
try to unlock the workspace. Leaving the workspace locked is a good
indication something went wrong and also prevents other changes from
being applied before the newest state is properly uploaded.

Additionally we now output the lock ID when a lock or force-unlock
action failed.

### CODE REVIEW

File: backend/remote/backend_state.go [hunk 1]
```diff
@@ -31,12 +32,12 @@ func (r *remoteClient) Get() (*remote.Payload, error) {
 			// If no state exists, then return nil.
 			return nil, nil
 		}
-		return nil, fmt.Errorf("Error retrieving remote state: %v", err)
+		return nil, fmt.Errorf("Error retrieving  state: %v", err)
```
Comment (pselle) [inline backend/remote/backend_state.go]: Nit: fyi you have an extra space here left where you removed `remote`

File: backend/remote/backend_state.go [hunk 2]
```diff
@@ -141,7 +153,12 @@ func (r *remoteClient) Unlock(id string) error {
 
 	// Verify the optional force-unlock lock ID.
 	if r.organization+"/"+r.workspace.Name != id {
-		lockErr.Err = fmt.Errorf("lock ID does not match existing lock")
+		lockErr.Err = fmt.Errorf(
```
Comment (pselle) [inline backend/remote/backend_state.go]: Nice additions making this error more informative!

File: backend/remote/backend_state.go (hunk 1 above)
Comment (svanharmelen) [reply backend/remote/backend_state.go]: Good catch! Let me update that!

Comment (svanharmelen) [reply backend/remote/backend_state.go]: Updated...

**Pull request merged**